File: sandwich_rust/src/storage_interface.rs
use crate::data_structures::sandwich::sandwich::Sandwich;
use crate::utils::PersistanceResponse;


pub trait StorageInterface {
        fn setup(file_path: String) -> Self;
        fn get_sandwich(&self, name: String) -> Sandwich;
        fn save_sandwich(&self, sandwich: Sandwich) -> PersistanceResponse;
        fn delete_sandwich(&self, sandwich: Sandwich) -> PersistanceResponse;
}

File: sandwich_go/src/storageinterface/storageinterface.go
package storageinterface

import (
	ds "sandwich_go/src/datastructures"
	ut "sandwich_go/src/utils"
)

type StorageInterface interface {
	Setup(file_path string) // to be used for setting up struct
	Get_sandwich(name string) ds.Sandwich
	Save_sandwich(sandwich ds.Sandwich) ut.Response
	Delete_sandwich(sandwich ds.Sandwich) ut.Response
}
